test/unit/FixedLatencyMemoryTest.cc

```diff
@@ -33,11 +33,68 @@ TEST(LatencyMemoryInterfaceTest, FixedWriteData) {
   EXPECT_TRUE(mmu->hasPendingRequests());
 
   // Tick once - request should still be pending
-  mmu->tick();
+  mem->tick();
   EXPECT_TRUE(mmu->hasPendingRequests());
 
   // Tick again - request should have completed
-  mmu->tick();
+  mem->tick();
+  EXPECT_FALSE(mmu->hasPendingRequests());
+
+  auto resp = mem->getUntimedData(0, 4);
+
+  uint32_t castedValue = 0;
+  memcpy(&castedValue, resp.data(), 4);
+  EXPECT_EQ(castedValue, 0xDEADBEEF);
+}
+
+TEST(FixedLatencyMemoryTest, UntimedWriteData) {
+  // Create a memory interface with a two cycle latency
+  std::shared_ptr<simeng::memory::Mem> mem =
+      std::make_shared<simeng::memory::FixedLatencyMemory>(4, 2);
+
+  std::vector<char> data = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9'};
+  uint8_t dataSize = 10;
+  mem->sendUntimedData(data, 0, dataSize);
+  auto payload = mem->getUntimedData(0, dataSize);
+  for (size_t i = 0; i < dataSize; i++) {
+    EXPECT_EQ(payload[i], data[i]);
+  }
+}
+
+// Test that we can write data and it completes after a number of cycles.
+TEST(FixedLatencyMemoryTest, ReadData) {
+  // Create a memory interface with a two cycle latency
+  std::shared_ptr<simeng::memory::Mem> mem =
+      std::make_shared<simeng::memory::FixedLatencyMemory>(4, 2);
+
+  VAddrTranslator fn = [](uint64_t addr, uint64_t pid) -> uint64_t {
+    return addr;
+  };
+
+  std::shared_ptr<simeng::memory::MMU> mmu =
+      std::make_shared<simeng::memory::MMU>(fn);
+
+  auto connection =
+      simeng::PortMediator<std::unique_ptr<simeng::memory::MemPacket>>();
+  auto port1 = mmu->initPort();
+  auto port2 = mem->initPort();
+  connection.connect(port1, port2);
+
+  EXPECT_FALSE(mmu->hasPendingRequests());
+
+  // Write a 32-bit value to memory
+  // Should ignore the 7 cycle latency and opt for the interface defined latency
```

Comment:
What 7 cycle latency?

Reply:
comment from FixedLatencyMemoryInterface tests, I have removed it